clients/ts/signalr/spec/LongPollingTransport.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { HttpResponse } from "../src/HttpClient";
+import { LogLevel } from "../src/ILogger";
+import { TransferFormat } from "../src/ITransport";
+import { NullLogger } from "../src/Loggers";
+import { LongPollingTransport } from "../src/LongPollingTransport";
+import { ConsoleLogger } from "../src/Utils";
+
+import { TestHttpClient } from "./TestHttpClient";
+import { asyncit as it, PromiseSource } from "./Utils";
+
+describe("LongPollingTransport", () => {
+    it("shuts down poll after timeout even if server doesn't shut it down on receiving the DELETE", async () => {
+        let firstPoll = true;
+        const pollCompleted = new PromiseSource();
+        const client = new TestHttpClient()
+            .on("GET", async (r) => {
+                if (firstPoll) {
+                    firstPoll = false;
+                    return new HttpResponse(200);
+                } else {
+                    // Turn 'onabort' into a promise.
+                    const abort = new Promise((resolve, reject) => r.abortSignal.onabort = resolve);
+                    await abort;
+
+                    // Signal that the poll has completed.
+                    pollCompleted.resolve();
+                    return new HttpResponse(204);
+                }
+            })
+            .on("DELETE", (r) => new HttpResponse(202));
+        const transport = new LongPollingTransport(client, null, NullLogger.instance, false, 100);
```

Comment:
Do we know this 100 ms timeout is honored because the default 5000 ms timeout will cause the test to timeout?

Reply:
Before I added this parameter, the test would flake out. Sometimes it would pass, sometimes it wouldn't, and it was based on the jasmine 5000 ms timeout. It's very consistent with this change so we know it should be honored.